Clarify usage of Distributed Tracing Extension by OpenTelemetry (#1)
dynatrace-oss-contrib/spec: extensions/distributed-tracing.md

# Distributed Tracing extension

This extension embeds context from

> [W3C TraceContext](https://www.w3.org/TR/trace-context/) into a CloudEvent.

Reviewer: Maybe add a link to https://cloudevents.io/?

Author: Hum it's already the CloudEvents specification, so I guess it's clear enough? 😅

> The [OpenTelemetry specification](

Reviewer 2: You could introduce OTel in one sentence with a link to https://opentelemetry.io.

Author: I changed it a bit, to try to say how this extension relates to OTel and what OTel is.

> https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/context.md#overview)

Reviewer 2: ```suggestion
https://github.com/open-telemetry/opentelemetry-specification/blob/v1.8.0/specification/context/context.md#overview)
```
Would make sure the link doesn't go stale as content might be shuffled around.

defines context as:

> A Context is a propagation mechanism which carries execution-scoped values across
 API boundaries and between logically associated execution units. Cross-cutting
 concerns access their data in-process using the same shared Context object.

Context propagation enables `Distributed Tracing`.

> Also from [OpenTelemetry specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/overview.md#tracing-signal):

Reviewer 2: ```suggestion
Also from the [OpenTelemetry specification](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.8.0/specification/overview.md#tracing-signal):
```

> A distributed trace is a set of events, triggered as a result of a single
 logical operation, consolidated across various components of an application.
 A distributed trace contains events that cross process, network and security boundaries.

## Using the Distributed Tracing Extension

The

> [OpenTelemetry Semantic Conventions for CloudEvents](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/cloudevents.md)

Reviewer 2: ```suggestion
[OpenTelemetry Semantic Conventions for CloudEvents](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.9.0/specification/trace/semantic_conventions/cloudevents.md)
```
(once released)

> defines in which scenarios this extension should be used and how to use it.

Reviewer 2: ```suggestion
define in which scenarios this extension should be used and how to use it.
```

# Distributed Tracing extension

This extension embeds context from
[W3C TraceContext](https://www.w3.org/TR/trace-context/) into a CloudEvent.
The goal of this extension is to offer means to carry context when instrumenting
CloudEvents based systems with OpenTelemetry.

The [OpenTelemetry](https://opentelemetry.io/) project is a collection
of tools, APIs and SDKs that can be used to instrument, generate, collect,
and export telemetry data (metrics, logs, and traces) to help you
analyze your software’s performance and behavior.

The OpenTelemetry specification defines both

> [Context](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.8.0/specification/context/context.md#overview)

Reviewer: Did you add the link to a certain version (`1.8.0`) on purpose? Maybe we should point to latest?

Author: Yes, from @arminru's comments. This is to ensure if things move around the links here don't break.
